logging/nxscope/nxscope_iser.c

```diff
@@ -0,0 +1,248 @@
+
+#include <nuttx/config.h>
+
+#include <debug.h>
+#include <fcntl.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <termios.h>
+#include <unistd.h>
+
+#include <logging/nxscope/nxscope.h>
+
+/****************************************************************************
+ * Pre-processor Definitions
+ ****************************************************************************/
+
+#ifndef CONFIG_SERIAL_TERMIOS
+#  error
+#endif
+
+/****************************************************************************
+ * Private Type Definition
+ ****************************************************************************/
+
+struct nxscope_intf_ser_s
+{
+  FAR struct nxscope_ser_cfg_s *cfg;
+  int                           fd;
+};
+
+/****************************************************************************
+ * Private Function Protototypes
+ ****************************************************************************/
+
+static int nxscope_ser_send(FAR struct nxscope_intf_s *intf,
+                            FAR uint8_t *buff, int len);
+static int nxscope_ser_recv(FAR struct nxscope_intf_s *intf,
+                            FAR uint8_t *buff, int len);
+
+/****************************************************************************
+ * Private Data
+ ****************************************************************************/
+
+struct nxscope_intf_ops_s g_nxscope_ser_ops =
+{
+  .send   = nxscope_ser_send,
+  .recv   = nxscope_ser_recv
+};
+
+/****************************************************************************
+ * Private Functions
+ ****************************************************************************/
+
+/****************************************************************************
+ * Name: nxscope_ser_send
+ ****************************************************************************/
+
+static int nxscope_ser_send(FAR struct nxscope_intf_s *intf,
+                            FAR uint8_t *buff, int len)
+{
+  FAR struct nxscope_intf_ser_s *priv = NULL;
+
+  DEBUGASSERT(intf);
+  DEBUGASSERT(intf->priv);
+
+  /* Get priv data */
+
+  priv = (FAR struct nxscope_intf_ser_s *) intf->priv;
+
+  /* Write data */
+
+  return write(priv->fd, buff, len);
+}
+
+/****************************************************************************
+ * Name: nxscope_ser_recv
+ ****************************************************************************/
+
+static int nxscope_ser_recv(FAR struct nxscope_intf_s *intf,
+                            FAR uint8_t *buff, int len)
+{
+  FAR struct nxscope_intf_ser_s *priv = NULL;
+  int                            ret  = OK;
+
+  DEBUGASSERT(intf);
+  DEBUGASSERT(intf->priv);
+
+  /* Get priv data */
+
+  priv = (FAR struct nxscope_intf_ser_s *) intf->priv;
+
+  /* Read data */
+
+  ret = read(priv->fd, buff, len);
+
+  if (ret < 0)
+    {
+      if ((priv->cfg->nonblock == true) && (errno == EAGAIN))
+        {
+          ret = 0;
```

Comment:
```suggestion
          ret = OK;
```

Reply:
We read 0 bytes, so I think 0 is more suitable here